pkg/webhook/cspc.go

```diff
@@ -262,11 +318,17 @@ var (
 	// Value of the keys --
 	// 1. In case of striped this is the minimum number of disk required.
 	// 2. In all other cases this is the exact number of disks required.
-	SupportedPRaidType = map[cstor.PoolType]int{
-		cstor.PoolStriped:  1,
-		cstor.PoolMirrored: 2,
-		cstor.PoolRaidz:    3,
-		cstor.PoolRaidz2:   6,
+	SupportedPRaidType = map[cstor.PoolType]validateRaidBDCount{
+		cstor.PoolStriped:  isStripedBDCountValid,
+		cstor.PoolMirrored: isMirroredBDCountValid,
+		cstor.PoolRaidz:    isRaidzBDCountValid,
+		cstor.PoolRaidz2:   isRaidz2BDCountValid,
+	}
+	// SupportedCompression is a map holding the supported compressions
+	SupportedCompression = map[string]bool{
+		"":    true,
+		"off": true,
+		"lz":  true,
```

Comment:
Not sure what and all are supporting
```c
 static zprop_index_t compress_table[] = {
                { "on",         ZIO_COMPRESS_ON },
                { "off",        ZIO_COMPRESS_OFF },
                { "lzjb",       ZIO_COMPRESS_LZJB },
                { "gzip",       ZIO_COMPRESS_GZIP_6 },  /* gzip default */
                { "gzip-1",     ZIO_COMPRESS_GZIP_1 },
                { "gzip-2",     ZIO_COMPRESS_GZIP_2 },
                { "gzip-3",     ZIO_COMPRESS_GZIP_3 },
                { "gzip-4",     ZIO_COMPRESS_GZIP_4 },
                { "gzip-5",     ZIO_COMPRESS_GZIP_5 },
                { "gzip-6",     ZIO_COMPRESS_GZIP_6 },
                { "gzip-7",     ZIO_COMPRESS_GZIP_7 },
                { "gzip-8",     ZIO_COMPRESS_GZIP_8 },
                { "gzip-9",     ZIO_COMPRESS_GZIP_9 },
                { "zle",        ZIO_COMPRESS_ZLE },
                { "lz4",        ZIO_COMPRESS_LZ4 },
                { NULL }
        };
```
cc: @sonasingh46

Reply:
@shubham14bajpai We need to support above all compressions.

Reply:
As of know, the existing map is good. We will revisit if we need to support more...

Reply:
Added a TODO comment for this. Once we are sure we can update the map.